client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/DeviceConnectionInfoCache.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.deviceconnection.infinispan.client;
+
+import io.opentracing.SpanContext;
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * A repository for keeping connection information about devices.
+ *
+ */
+public interface DeviceConnectionInfoCache {
```

Comment:
This name very much implies there being only one cache for the device connection data. However, for #1272 there will be other data added (mapping between device id and id of the protocol adapter instance that has a command consumer for that device). I would rather put this data in a separate Infinispan cache. So then there would be 2 Infinispan caches used for device connection data.

So, for now the name is Ok, for #1272 I think it will better be renamed to something like `LastKnownGatewayInfoCache`.